dump/dump_test.go

```diff
@@ -0,0 +1,456 @@
+/*
+Copyright 2017 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dump
+
+import (
+	"testing"
+)
+
+func TestSdumpPrimitives(t *testing.T) {
+	testcases := []struct {
+		name     string
+		input    func() interface{}
+		expected string
+	}{
+		{
+			"boolTrue",
+			func() interface{} {
+				return true
+			},
+			"var Obj = true\n",
+		},
+		{
+			"boolFalse",
+			func() interface{} {
+				return false
+			},
+			"var Obj = false\n",
+		},
+		{
+			"int",
+			func() interface{} {
+				return 10
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"int8",
+			func() interface{} {
+				return int8(10)
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"int16",
+			func() interface{} {
+				return int16(10)
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"int32",
+			func() interface{} {
+				return int32(10)
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"int64",
+			func() interface{} {
+				return int64(10)
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"uint",
+			func() interface{} {
+				return uint(10)
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"uint8",
+			func() interface{} {
+				return uint8(10)
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"uint16",
+			func() interface{} {
+				return uint16(10)
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"uint32",
+			func() interface{} {
+				return uint32(10)
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"uint64",
+			func() interface{} {
+				return uint64(10)
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"float32",
+			func() interface{} {
+				return float32(10.5)
+			},
+			"var Obj = 10.5\n",
+		},
+		{
+			"float64",
+			func() interface{} {
+				return float64(10.5)
+			},
+			"var Obj = 10.5\n",
+		},
+		{
+			"complex64",
+			func() interface{} {
+				return complex64(10 + 10.5i)
+			},
+			"var Obj = complex64(10+10.5i)\n",
```

Comment:
You could just do this for all the primitive types, and then they'd be preserved.  Or am I missing something?